hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -271,6 +296,8 @@ extends Importer:
       Term.Ret(term(body))
     case Modified(Keyword.`do`, kwLoc, body) =>
       Term.Blk(term(body) :: Nil, unit)
+    case TypeDef(Mod, head, _, _) =>
+      Term.Mod(term(head))
```

Comment:
Do we have examples of this in the tests? And why is it restricted to modules? (In JS, one can also define and return classes.) Anyway, I don't see a good reason for this. Feels like  term `(module M with { ... })` should be desugared to `{module M with { ... }; M}`. You don't need a new `Mod` term form.

Reply:
I don't really understand your question, but I guess it's related to this comment: https://github.com/hkust-taco/mlscript/pull/233#discussion_r1850475848.

The reason of having this hack is that I can't figure out a good way to parse a module declaration/definition and a module modifier into different `Tree` things, so all of them, including `module` modifiers, get parsed into `TypeDef`. 

I added a case here to elaborate module modifiers into `Term.Mod` so that it can be further processed.

Reply:
Indeed, this was related to my previous assumption that `Mod` meant something different.

> all of them, including `module` modifiers, get parsed into `TypeDef`

BTW there is absolutely no problem with this. Our parsing philosophy is that we parse any tree that makes visual sense (such as module definitions), in any position, and detangle things later in the elaboration pass. This should allow us to have much more specific and helpful error messages while ensuring visual consistency of the syntax.

Reply:
BUT you **must** check that this module `TypeDef` does **not** have any additional things inside, instead of silently discarding them using `_`. Think about the user experience of doing that.